app/importers/importers/importer.rb

```diff
@@ -192,5 +206,8 @@ def self.without_auto_timestamps
       Kithe::Model.record_timestamps = original
     end
 
+    def same_friendlier_id_different_type
+      Kithe::Model.where(friendlier_id:metadata['id']).where.not(type:self.class.destination_class.to_s).first
+    end
```

Comment:
I don't like that this is doing an extra SQL fetch for every single imported record. 

It seems like there should be a way to handle this (unusual) case without an extra fetch for every single object to import. 

Two possible approaches:

1. If the current code has a problem where it gets an error on _save_, because of the conflicting friendlier_id -- rescue that error, and recover from it *there*. So the extra Database work only happens when there really is a conflict. 

OR

2. If right now when looking for an existing work, we're doing `Work.find_by_friendlier_id`, you can change that code to `Kithe::Model.find_by_friendlier_id` -- the object will *still come back* as class `Work` if it is one. Then, at _that_ point, you can check `object_that_came_back.kind_of?(ClassIExpect)`, and if it's _not_, do the extra work there again (to skip or delete it whatever). 

Either of these two approaches should make it possible to handle conflicts (including by respecting a choice of either "skip" or "delete" strategies), without having to do an extra SQL fetch for every single import target.

Reply:
Yeah, good point. Option 2 is nice; will do that.